circuit_knitting/cutting/cutting_decomposition.py

```diff
@@ -282,3 +295,261 @@ def decompose_observables(
     }
 
     return subobservables_by_subsystem
+
+
+def generate_cutting_experiments(
+    circuits: QuantumCircuit | dict[str | int, QuantumCircuit],
+    observables: PauliList | dict[str | int, PauliList],
+    num_samples: int | float,
+) -> tuple[
+    list[QuantumCircuit] | dict[str | int, list[QuantumCircuit]],
+    list[tuple[float, WeightType]],
+]:
+    """
+    Generate cutting subexperiments and their associated weights.
+
+    If the input circuit and observables are not split into multiple partitions, the output
+    subexperiments will be contained within a 1D array.
+
+    If the input circuit and observables is split into multiple partitions, the output
+    subexperiments will be returned as a dictionary which maps a partition label to to
+    a 1D array containing the subexperiments associated with that partition.
+
+    In both cases, the subexperiment lists are ordered as follows:
+        :math:`[sample_{0}observable_{0}, sample_{0}observable_{1}, ..., sample_{0}observable_{N}, ..., sample_{M}observable_{N}]`
```

Comment:
I think this is fine as a first step, but I would really like to move away from this and instead expose an explicit map that allows looking up each (sample, observable) from the index in this flat list and vice versa.  The reason is because some experiments are actually redundant (#262) or potentially even unnecessary, i.e. if the only observables in that subregion are all Pauli `I`s (in which case those experiments can be skipped altogether).

Reply:
Ya let's think about this more. If you do the dictionary you're back in a situation where you're either asking the user to set up their experiments for the primitive or you're providing a helper function to do it for them. As it is now, users can pass in the outputs of `decompose` straight into the primitive and get their results, which are the same shape as the input.

There is probably some middle ground we can arrive on if we think on it longer.

Reply:
Deleted a comment bc I misunderstood your statement here. Yes, this sounds like a good idea